### PR TITLE
Drop unused structure "chain_entry_t"

### PR DESCRIPTION
**Pull request by vacantron**
This structure was out of date and might become buggy in the upcoming system simulation.

### CODE REVIEW

File: src/cache.c [hunk 1]
```diff
@@ -196,33 +196,58 @@ void *cache_get(const cache_t *cache, uint32_t key, bool update)
...
+    lfu_entry_t *replaced_entry = NULL, *entry;
+    hlist_for_each_entry (entry, &cache->map->ht_list_head[cache_hash(key)],
+                          ht_list) {
+        if (entry->key != key)
+            continue;
+        /* update the existing cache */
+        if (entry->value != value) {
+            replaced_entry = entry;
+            break;
+        } else {
```
Comment (jserv) [inline src/cache.c]: Remove `else` statement since its previous line is exactly `break`.

File: src/emulate.c [hunk 2]
```diff
@@ -533,8 +533,40 @@ FORCE_INLINE bool insn_is_unconditional_branch(uint8_t opcode)
...
+    case rv_insn_jalr:
+#if RV32_HAS(EXT_C)
+    case rv_insn_cjalr:
+    case rv_insn_cjr:
+#endif
+        return true;
+    default:
+        return false;
+    }
+    __UNREACHABLE;
```
Comment (jserv) [inline src/emulate.c]: It is not necessary to place `__UNREACHABLE` here.

File: src/emulate.c [hunk 3]
```diff
@@ -533,8 +533,40 @@ FORCE_INLINE bool insn_is_unconditional_branch(uint8_t opcode)
...
+    case rv_insn_jal:
+#if RV32_HAS(EXT_C)
+    case rv_insn_cjal:
+    case rv_insn_cj:
+#endif
+        return true;
+    default:
+        return false;
+    }
+    __UNREACHABLE;
```
Comment (jserv) [inline src/emulate.c]: Ditto. Omit `__UNREACHABLE` here.

**Review by jserv — COMMENTED** [review]
# Benchmarks

<details>

| Benchmark suite | Current: 4b32e328760e3bb5bae5682774c309f66390ed26 | Previous: 1a12fd62ffd3cd6a86bcd608f52e337a2fb279cf | Ratio |
|-|-|-|-|
| `Dhrystone` | `1542` Average DMIPS over 10 runs | `1522` Average DMIPS over 10 runs | `0.99` |
| `Coremark` | `1439.724` Average iterations/sec over 10 runs | `1403.675` Average iterations/sec over 10 runs | `0.97` |

</details>

This comment was automatically generated by [workflow](https://github.com/sysprog21/rv32emu/actions?query=workflow%3ABenchmark) using [github-action-benchmark](https://github.com/marketplace/actions/continuous-benchmark).

**Comment by vacantron** [comment]
@jserv , could you re-run the jobs from the tab of GitHub Actions? It seems like the Internet issue that causes `make artifact` failed.

**Comment by jserv** [comment]
Thank @vacantron for contributing!

**Pull request merged**